lldb/source/Commands/CommandObjectTarget.cpp

```diff
@@ -2547,6 +2561,7 @@ class CommandObjectTargetModulesDumpSeparateDebugInfoFiles
 
     OptionValueBoolean m_json = false;
     OptionValueBoolean m_errors_only = false;
+    OptionValueBoolean m_load_all_debug_info = true;
```

Comment:
I wonder if it would be better to disable loading all of the debug info by default. Then instead of adding the `--defer-load-all-debug-info` flag we could add a flag like `--force-load-all-debug-info` that would force loading it.

It seems a bit odd to me that a `dump` command would modify state like this by default by loading up the debug info instead of reporting the current state. But this would be a behavior change from the current lldb so maybe not desirable.

@JDevlieghere @jeffreytan81 @clayborg any thoughts on this?